src/xhr.js

```diff
@@ -1,31 +1,80 @@
 import { fetch } from './fetch.js';
 import { Evented } from './evented.js';
 
+// Extends from Evented to fire events
 export class XMLHttpRequest extends Evented {
-  open (method, url) {
+  constructor() {
+    super();
+    this.headers = {};
+    this.readyState = 1;
+    this.withCredentials = true;
+  }
+
+  // Stores method and url
+  open(method, url) {
     this.method = method;
     this.url = url;
   }
-  send () {
-    // print('XHR', this.method, this.url);
-    fetch(this.url, { method: this.method })
+
+  /**
+   * Resets status and values
+   * Needed by some implementations
+   */
+  abort() {
+    this.responseText = '';
+    this.response = null;
+    this.readyState = 0;
+    this.status = 0;
+  }
+
+  // Append header to headers object
+  setRequestHeader(...args) {
+    this.headers[args[0]] = args[1];
+  }
+
+  // Returns all headers
+  getAllResponseHeaders() {
+    return this.responseHeaders;
+  }
+
+  /**
+   * Uses fetch to fire request
+   * Fires both load and readystatechange so registered components know data was loaded
+   */
+  send(body) {
+    fetch(this.url, { method: this.method, body })
       .then(res => {
+        this.readyState = 4;
```

Comment:
Can you define the corresponding constants instead of using numbers directly?

value | name | description
-- | -- | --
0 | UNSENT | Client has been created. open() not called yet.
1 | OPENED | open() has been called.
2 | HEADERS_RECEIVED | send() has been called, and headers and status are available.
3 | LOADING | Downloading; responseText holds partial data.
4 | DONE | The operation is complete.

Reply:
Sure, I can define 'DONE' if you want to, as it's the only one we'll be using, all others are not necessary due to our fetch implementation

Reply:
we use UNSENT and OPENED in `abort` and `constructor` as well